js_modules/dagster-ui/packages/ui-core/src/asset-graph/AssetNodeFacetSettingsButton.tsx

```diff
@@ -0,0 +1,164 @@
+import {useEffect, useState} from 'react';
+import {AssetNodeFacet, AssetNodeFacetDefaults} from './AssetNodeFacets';
+import {Box, Button, Colors, Dialog, DialogFooter, Icon} from '@dagster-io/ui-components';
+import {AssetNodeFacetsPicker} from './AssetNodeFacetsPicker';
+import {AssetNode2025} from './AssetNode2025';
+import {
+  AssetNodeFragmentBasic,
+  LiveDataForNodeMaterializedAndStaleAndOverdue,
+  LiveDataForNodeMaterializedWithChecks,
+} from './__fixtures__/AssetNode.fixtures';
+import {LiveDataForNodeWithStaleData, tokenForAssetKey} from './Utils';
+import {
+  AssetCheckExecutionResolvedStatus,
+  AssetCheckSeverity,
+  buildAssetCheck,
+  buildAssetCheckEvaluation,
+  buildAssetCheckExecution,
+  buildAssetKey,
+  buildAssetNode,
+  buildMaterializationEvent,
+  buildStaleCause,
```

Comment:
Hmm, I'm torn about whether we should use the builders in production code. While convenient I think there is potential of unintended changes as the schema changes. Also they tend to add more fields than necessary. Lastly I worry about bundle size since they depend on each other recursively.

Reply:
Oh interesting I think you're right that this impacts bundle size quite a bit.

This is meant to be a "preview" of what your asset nodes will look like so it's actually in the app. Will de-couple it from these builders